src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = Storage.getFruitStorage();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+
+            handler.apply(storage, transaction);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
+        return new HashMap<>(storage);
+    }
+}
```

Comment:
The ShopServiceImpl class is placed in the default package. According to the checklist, you should organize your classes into packages based on their responsibilities. For example, this class could be placed in a `service.impl` package as it is an implementation of a service.